src/api/app/controllers/public_controller.rb

```diff
@@ -95,6 +95,14 @@ def project_index
     pass_to_backend path
   end
 
+  # GET /public/source
+  def projects
+    @projects = Rails.cache.fetch(['projectlist', Project.maximum(:updated_at), User.current.forbidden_project_ids]) do
+      @projects = Project.pluck(:name).sort
+    end
+    render 'source/projects'
+  end
```

Comment:
Directly routing won't work because of the before filter :(
Yeah, already thought about that!
Let's wait for the response from @digitaltom anyway...